src/Registry.php

```diff
@@ -508,11 +508,13 @@ public function set($path, $value, $separator = null)
 		switch (true)
 		{
 			case (is_object($node)):
-				$result = $node->{$nodes[$i]} = $value;
+				$result = $node->{$nodes[$i]};
```

Comment:
This needs to be `$result = $node->{$nodes[$i]} ?? null;` (similar for the array case), as is it's causing unit test failures because of undefined properties.